src/object.cr

```diff
@@ -219,7 +219,14 @@ class Object
   end
 
   # Returns `self`.
+  #
   # `Nil` overrides this method and raises `NilAssertionError`, see `Nil#not_nil!`.
+  #
+  # This method can be used to remove `Nil` from a union type.
+  # However, it should be avoided if possible and is often considered a code smell.
+  # Usually, you can write code in a way that the compiler can safely exclude `Nil` types,
+  # for example using [`if var`](https://crystal-lang.org/reference/syntax_and_semantics/if_var.html`).
+  # `not_nil!` is only meant as a last resort when no other solution is possible.
```

Comment:
```suggestion
  # `not_nil!` is only meant as a last resort when there's no other way to explain this to the compiler.
  # Either way, consider instead raising a concrete exception with a descriptive message.
```

Reply:
I'll point out just in case (it's hard to see): my suggestion also adds the whole line
`# Either way, consider instead raising a concrete exception with a descriptive message.`